File: uxmobile/src/main/java/sk/uxtweak/uxmobile/model/event/OrientationEvent.java
package sk.uxtweak.uxmobile.model.event;

import org.json.JSONArray;
import org.json.JSONException;

/**
 * Created by matej on 24.10.2017.
 */

public class OrientationEvent extends Event {
    protected static final int INDEX_ORIENTATION = 2;

    private final int mOrientation;

    public OrientationEvent(long startTime, int orientation) {
        super(startTime);
        mOrientation = orientation;
    }

    @Override
    protected String getType() {
        return TYPE_ORIENTATION;
    }

    @Override
    public JSONArray toJson() throws JSONException {
        return super.toJson()
                .put(INDEX_ORIENTATION, mOrientation);
    }
}

File: uxmobile/src/main/java/sk/uxtweak/uxmobile/model/event/OrientationEvent.kt
package sk.uxtweak.uxmobile.model.event

import org.json.JSONArray

class OrientationEvent(startTime: Long, private val orientation: Int) : Event(startTime) {
    override fun getType() = TYPE_ORIENTATION

    override fun toJson(): JSONArray = super.toJson()
        .put(INDEX_ORIENTATION, orientation)

    companion object {
        const val INDEX_ORIENTATION = 2
    }
}
